core/src/main/java/cz/o2/proxima/storage/watermark/GlobalWatermarkThroughputLimiter.java

```diff
@@ -122,6 +123,7 @@ public void close() {
     if (tracker != null) {
       ExceptionUtils.unchecked(tracker::close);
     }
+    lastGlobalWatermarkUpdate = Long.MIN_VALUE;
```

Comment:
what is this good for? we don't reuse instances right?

Reply:
We were reusing them, but we are cloning them instead now. The reuse was not thread-safe. This is a left-over.

Reply:
Fixed.